website/routes.py

```diff
@@ -70,6 +71,19 @@ def set_status_message(user):
             extra={}
         )
 
+def anonymize_ip(request):
+    remote_addr = request.remote_addr
+    if remote_addr:
+        try:
+            ip = ipaddress.ip_address(unicode(remote_addr))
+        except ipaddress.AddressValueError:
+            return None
+        if ip.version == 4:
+            return '.'.join(ip.exploded.split('.')[:-1]) + '.0'
+        elif ip.version == 6:
+            return ':'.join(ip.exploded.split(':')[:-1])[:-1] + '0:0000'
```

Comment:
I don't know how error prone this IP splitting is going to be:
Here is what I'm trying to do:
 IPv4: 12.214.31.144 -> 12.214.31.0
IPv6: 2001:0db8:0000:0000:0000:ff00:0042:8329 -> 2001:0db8:0000:0000:0000:ff00:0040:0000

Reply:
Probably best for this to be moved to a utility module.

Reply:
This is a perfect opportunity to use a bitwise `&` operator. You'll be able to avoid any ambiguity with the string parse, for example:

```python
In [1]: ipv4 = ipaddress.ip_address(unicode('255.255.255.255'))

In [2]: ipv6 = ip_address(unicode('ffff:ffff:ffff:ffff:ffff:ffff:ffff:ffff'))

In [3]: ip_address(int(ipv4) & 0xFFFFFF00).exploded
Out[3]: u'255.255.255.0'

In [4]: ip_address(int(ipv6) & 0xFFFFFFFFFFFF00000000000000000000).exploded
Out[4]: u'ffff:ffff:ffff:0000:0000:0000:0000:0000'
```

Reply:
cool @icereval!